app/url/urlController.js

```diff
@@ -0,0 +1,104 @@
+const DeleteUrlHttpRequest = require('./delete-url/deleteUrlHttpRequest');
+
+class UrlRouterBuilder {
+    constructor ({express, urlRepository, urlFactory, getUrlResponseBuilder, searchUrlResponseBuilder, pasteRepository, jwt, createUrlResponseBuilder, idGenerator}) {
+        this.router = express.Router();
+        this.urlRepository = urlRepository;
+        this.urlFactory = urlFactory;
+        this.getUrlResponseBuilder = getUrlResponseBuilder;
+        this.searchUrlResponseBuilder = searchUrlResponseBuilder;
+        this.pasteRepository = pasteRepository;
+        this.jwt = jwt;
+        this.createUrlResponseBuilder = createUrlResponseBuilder;
+        this.idGenerator = idGenerator;
+    }
+
+    createRoutes() {
+        this.router.get('/:pasteId', async (request, response) => {
+            const validator = new GetUrlValidator();
+            const presenter = new HttpPresenter(request, response);
+            const interactor = new GetUrlInteractor({
+                presenter, 
+                validator, 
+                responseBuilder: this.getUrlResponseBuilder, 
+                urlRepository: this.urlRepository
+            });
+
+            try {
+                await interactor.execute(new GetUrlHttpRequest(request));
+            } catch (error) {
+                presenter.presentFailure(error);
+            }
+        });
+
+        this.router.post('/', async (request, response) => {
+            const validator = new CreateUrlValidator();
+            const presenter = new HttpPresenter(request, response);
+            const interactor = new CreateUrlInteractor({
+                presenter, 
+                validator, 
+                urlFactory: this.urlFactory, 
+                urlRepository: this.urlRepository, 
+                pasteRepository: this.pasteRepository, 
+                jwt: this.jwt, 
+                responseBuilder: this.createUrlResponseBuilder, 
+                idGenerator: this.idGenerator
+            });
+
+            try {
+                await interactor.execute(new CreateUrlHttpRequest(request));
+            } catch (error) {
+                presenter.presentFailure(error);
+            }
+        });
+
+        this.router.post('/search-url', async (request, response) => {
+            const validator = new SearchUrlValidator();
+            const presenter = new HttpPresenter(request, response);
+            const interactor = new SearchUrlInteractor({
+                presenter, 
+                validator, 
+                urlRepository: this.urlRepository, 
+                responseBuilder: this.searchUrlResponseBuilder
+            });
+
+            try {
+                await interactor.execute(new SearchUrlHttpRequest(request));
+            } catch (error) {
+                presenter.presentFailure(error);
+            }
+        });
+
+        this.router.delete('/:pasteId', async (request, response) => {
```

Comment:
Same here)

Reply:
Btw, I'm not sure if we ever need this route. Or any, except of "create new url"